core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/offer/service/processor/AbstractBaseProcessor.java

```diff
@@ -279,6 +283,18 @@ protected boolean couldOrderItemMeetOfferRequirement(OfferItemCriteria criteria,
 
         return appliesToItem;
     }
+
```

Comment:
Would it make sense to pull this out into a helper class?  Are there other places we could use this logic.  If offer rules have this issue, what about Customer Segment, Targeters, other rule builders?